compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingInterface.cpp

```diff
@@ -266,15 +278,238 @@ static LogicalResult replaceAllStoresWithTiledVersion(
+/// `boundedSizesForLoops` have the same size.
 struct TilingResult {
   Operation *tiledOp;
-  SmallVector<scf::ForOp> loops;
+  // BitVector for the tiled loops in the loop range
   llvm::SmallBitVector tiledLoops;
+  // offsets for the input loop range
   SmallVector<OpFoldResult> tileOffsets;
+  // tile sizes for the input loop range
   SmallVector<OpFoldResult> tileSizes;
+  // tile sizes requested for the input loop range
+  SmallVector<Value> requestedTileSizes;
+
+  // The result information
+  //
+  // tiled loops.
+  SmallVector<scf::ForOp> loops;
+  // tile sizes for the resulted loops
+  SmallVector<Value> tileSizesForLoops;
+  // bounded tile size for the resulted loops
+  SmallVector<OpFoldResult> boundedSizesForLoops;
+
+  void dump() {
+    llvm::dbgs() << "<TilingResult>\n"
+                 << "tiledOp =\n";
+    if (tiledOp) {
+      tiledOp->dump();
+      llvm::dbgs() << "\n";
+    } else {
+      llvm::dbgs() << "nullptr\n";
+    }
+
+    llvm::dbgs() << "\ntileOffsets:\n";
+    for (auto tileOffset : tileOffsets) {
+      tileOffset.dump();
+    }
+
+    llvm::dbgs() << "\ntileSizes:\n";
+    for (auto tileSize : tileSizes) {
+      tileSize.dump();
+    }
+
+    llvm::dbgs() << "\nrequestedTileSizes:\n";
+    for (auto size : requestedTileSizes) {
+      size.dump();
+    }
+
+    llvm::dbgs() << "# of loops = " << loops.size() << "\n";
+    if (!loops.empty()) {
+      loops[0].dump();
+    }
+
+    llvm::dbgs() << "\ntileSizesForLoops:\n";
+    for (auto size : tileSizesForLoops) {
+      size.dump();
+    }
+
+    llvm::dbgs() << "\nboundedSizesForLoops:\n";
+    for (auto size : boundedSizesForLoops) {
+      size.dump();
+    }
+    llvm::dbgs() << "</TilingResult>\n";
+  }
 };
 
+/// Get the integer tile sizes from the requestedTileSizes
+static FailureOr<SmallVector<int64_t>> getTileSizesAsInt64(
+    ArrayRef<Value> requestedTileSizes) {
+  SmallVector<int64_t> tileSizes;
+
+  for (auto val : requestedTileSizes) {
+    if (auto constIndexOp = val.getDefiningOp<arith::ConstantIndexOp>()) {
+      auto v = constIndexOp.value();
+      // When tiling is not defined, let's use 1 here.
+      tileSizes.push_back(v == 0 ? 1 : v);
+    } else {
+      return failure();
+    }
+  }
+  return tileSizes;
+}
+
+static SmallVector<AffineMinOp> collectTileSizeMinOps(
+    ArrayRef<OpFoldResult> tileSizes) {
+  SmallVector<AffineMinOp> res;
+
+  for (auto ofr : tileSizes) {
+    if (auto value = ofr.dyn_cast<Value>()) {
+      if (auto minOp = value.getDefiningOp<AffineMinOp>()) {
+        res.push_back(minOp);
+      } else {
+        res.push_back(AffineMinOp());
+      }
+    } else {
+      res.push_back(AffineMinOp());
+    }
+  }
+  return res;
+}
+
+static SmallVector<scf::ForOp> collectNestedLoops(scf::ForOp outerLoop) {
+  SmallVector<scf::ForOp> loops;
+
+  outerLoop.getOperation()->walk<WalkOrder::PreOrder>(
+      [&](scf::ForOp op) { loops.push_back(op); });
+
+  return loops;
+}
```

Comment:
is this used? I can't find a use

Reply:
Oh, it became dead after going several refactoring passes. I will remove it. Thanks for catching it!